RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -159,11 +233,17 @@ void L1TJetsMatching<T>::produce(edm::StreamID iSId, edm::Event& iEvent, const e
   }
 
   output = categorise(*pfMatchedJets, pt1Min_, pt2Min_, pt3Min_, mjjMin_);
-  unique_ptr<std::vector<T>> output1(new std::vector<T>(output.first));
-  unique_ptr<std::vector<T>> output2(new std::vector<T>(output.second));
+  unique_ptr<std::vector<T>> output1(new std::vector<T>(std::get<0>(output)));
+  unique_ptr<std::vector<T>> output2(new std::vector<T>(std::get<1>(output)));
+  unique_ptr<std::vector<T>> output3(new std::vector<T>(std::get<2>(output)));
+  unique_ptr<std::vector<T>> output4(new std::vector<T>(std::get<3>(output)));
+  unique_ptr<std::vector<T>> output5(new std::vector<T>(std::get<4>(output)));
 
   iEvent.put(std::move(output1), "TwoJets");
   iEvent.put(std::move(output2), "ThreeJets");
+  iEvent.put(std::move(output3), "FourJets");
+  iEvent.put(std::move(output4), "FiveJets");
+  iEvent.put(std::move(output5), "SixJets");
```

Comment:
This does not follow the suggestion in https://github.com/cms-sw/cmssw/pull/41004#issuecomment-1464934311.

The idea is to leave the current behaviour of the module unchanged, and add a parameter (e.g. `selectionMode`) to steer the behaviour of the plugin, along the lines of the following pseudo-code
```cpp
if (selectionMode_ == "VBF") { // default behaviour

  output = categoriseVBF(*pfMatchedJets, pt1Min_, pt2Min_, pt3Min_, mjjMin_);
  unique_ptr<std::vector<T>> output1(new std::vector<T>(output.first));
  unique_ptr<std::vector<T>> output2(new std::vector<T>(output.second));
  iEvent.put(std::move(output1), "TwoJets");
  iEvent.put(std::move(output2), "ThreeJets");
}
else if (selectionMode_ == "VBFPlus2CentralJets") {

  output = categoriseVBFPlus2CentralJets(*pfMatchedJets, pt1Min_, pt2Min_, pt3Min_, mjjMin_);
  unique_ptr<std::vector<T>> output1(new std::vector<T>(std::get<0>(output)));
  unique_ptr<std::vector<T>> output2(new std::vector<T>(std::get<1>(output)));
  unique_ptr<std::vector<T>> output3(new std::vector<T>(std::get<2>(output)));

  iEvent.put(std::move(output1), "FourJets");
  iEvent.put(std::move(output2), "FiveJets");
  iEvent.put(std::move(output3), "SixJets");
}
```

Reply:
Flag is now introduced, and preserves original functionality when default value is used. Re-tested in the context of VBF parking studies with no apparent changes in the results